sdk/go/common/workspace/plugins.go

```diff
@@ -83,20 +88,36 @@ type pluginDownloadURLOverride struct {
 // pluginDownloadOverrideArray represents an array of overrides.
 type pluginDownloadOverrideArray []pluginDownloadURLOverride
 
-// get returns the URL and true if name matches an override's regular expression,
-// otherwise an empty string and false.
-func (overrides pluginDownloadOverrideArray) get(name string) (string, bool) {
+// get returns the URL and true if url matches an override's regular expression,
+// otherwise an empty string and false. The input url may contain placeholders
+// that match regular expression's named subgroups, the placeholders will be
+// replaced by matches values.
+func (overrides pluginDownloadOverrideArray) get(url string) (string, bool) {
 	for _, override := range overrides {
-		if override.reg.MatchString(name) {
-			return override.url, true
+		if match := override.reg.FindStringSubmatch(url); match != nil {
+			result := override.url
+			for i, name := range override.reg.SubexpNames() {
+				if i != 0 && name != "" {
```

Comment:
Do we want to support $index (e.g. $1) matches here as well? Do we want to consider allowing $0 so users could redirect to something like 'artefacts.internal.corp?pulumi=$0' to pass the whole original URL to their backend?

Reply:
Do you have a scenario in mind that is not possible with current implementation but would be possible with these suggestions? Or is it syntactic sugar that makes some overrides simpler?

> 'artefacts.internal.corp?pulumi=$0' to pass the whole original URL to their backend

This looks possible by capturing the original URLs with named groups and then repeating (re-building) them in place of $0.

The change is quite simple to make but maybe it also has some cost in terms of ambiguity and future backwards-compat? Not feeling strongly either way.

Reply:
I don't think it changes what's possible, just thinking about keeping this similar to other regex replace systems. I don't feel that strongly about it, if we want to just encourage names groups for users to keep things readable that seems ok.

Reply:
I implemented this now